pramen/core/src/test/scala/za/co/absa/pramen/core/mocks/SqlGeneratorDummy.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.pramen.core.mocks
+
+import za.co.absa.pramen.api.sql.{SqlConfig, SqlGenerator}
+
+import java.time.LocalDate
+
+class SqlGeneratorDummy(sqlConfig: SqlConfig) extends SqlGenerator {
+  def getSqlConfig: SqlConfig = sqlConfig
+
+  override def getDtable(sql: String): String = null
```

Comment:
You may consider using `???` instead of `null` to avoid the usage of `null`s in Scala.

Reply:
Interesting idea. I always use `null` in dummy classes, coming from Uncle Bob mocking principles. And I use `???` when something not implemented yet, but I need the project to compile.

The difference for me is:
- When `???` is returned it means the implementation should be there soon, and no `???` should be in PRs.
- When `null` is returned it means it is a dummy class and the implementation of this method is never intended.

Let me know if given this, you still prefer `???`.